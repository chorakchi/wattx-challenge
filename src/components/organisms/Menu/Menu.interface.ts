import { HTMLAttributes } from "react";
import { IconName } from './../../atoms/SVG/IconName.type';

export type menuItems = {
  icon: IconName;
  position?: string;
  href?: string;
  uri?: string;
  label?: string
};

export interface MenuProps extends HTMLAttributes<HTMLElement> {
  items: Array<menuItems>
  onChangeRoute?: (path: string) => void;
  currentPath?: string;
}
