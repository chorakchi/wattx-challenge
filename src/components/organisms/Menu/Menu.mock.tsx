export const mockData = {
  items: [
    {
      label: "Market overview",
      icon: "home",
      uri: "/",
    },
    {
      label: "Liquidity analysis page",
      icon: "analysis",
      uri: "/liquidity",
    },
    // {
    //   label: 'setting',
    //   icon:'setting',
    //   uri:'/setting'
    // },
    // {
    //   label: 'Log out',
    //   icon:'logout',
    //   uri:'/logout',
    // },
    {
      label: "info",
      icon: "info",
      uri: "https://github.com/WATTx/code-challenges",
      position: "end",
    },
  ],
};
