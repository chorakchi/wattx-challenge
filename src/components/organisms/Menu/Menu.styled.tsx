import styled, { css } from "styled-components";
import { Box } from "../../atoms";

export const StyledWrapper = styled(Box)`
  min-height: 500px;
  height: 100%;
  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 0;
  padding: 23px 0;

  @media screen and (max-width: 1280px) {
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    flex-direction: row;
    height: 72px;
    z-index: 2;
    width: 100%;
    min-height: 72px;
    padding: 0;
  }
`;

export const StyledItem = styled.div`
  width: 72px;
  padding: 23px 0;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  :hover {
    background: #0071ff38;
    color: #0071ff;
    box-shadow: 0px 5px 21px 2px rgba(28, 168, 255, 0.1);
  }
  ${({ selected }) =>
    selected
      ? css`
          cursor: default;
          background: #0071ff;
          color: white;
          :hover {
            background: #0071ff;
            color: white;
          }
        `
      : css`
          background: unset;
          color: #0071ff;
        `};
  transition: all 100ms ease-in-out;
`;

export const StyledGap = styled.div`
  flex: 1;
`;
