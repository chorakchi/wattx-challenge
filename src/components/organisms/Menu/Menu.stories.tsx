import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Menu } from "./Menu";
import { mockData } from "./Menu.mock";

export default {
  title: "Organisms/Menu",
  component: Menu,
} as ComponentMeta<typeof Menu>;

const Template: ComponentStory<typeof Menu> = (args) => <Menu {...args} />;

export const Menu_ = Template.bind({});
Menu_.args = {
  onSelect: (id) => {
    alert(id);
  },
  ...mockData,
};
