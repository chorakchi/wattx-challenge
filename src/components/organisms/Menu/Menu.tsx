import { StyledWrapper, StyledItem, StyledGap } from "./Menu.styled";
import { MenuProps } from "./Menu.interface";
import { SVG } from "components";

export const Menu: React.FC<MenuProps> = ({
  items,
  onChangeRoute,
  currentPath,
  ...props
}: MenuProps) => {
  const menuItem = items.filter((item) => item.position != "end");
  const menuItemSecondPart = items.filter((item) => item.position === "end");
  return (
    <StyledWrapper {...props}>
      {menuItem.map((item) => (
        <StyledItem
          onClick={() => item.uri && onChangeRoute(item.uri)}
          selected={currentPath ? currentPath == item.uri : "/" == item.uri}
        >
          <SVG size="medium" name={item.icon} />
        </StyledItem>
      ))}
      <StyledGap />
      {menuItemSecondPart.map((item) => (
        <StyledItem
          href={item.href}
          onClick={() => item.uri && onChangeRoute(item.uri)}
          selected={item.uri && currentPath == item.uri}
        >
          <SVG size="medium" name={item.icon} />
        </StyledItem>
      ))}
    </StyledWrapper>
  );
};
