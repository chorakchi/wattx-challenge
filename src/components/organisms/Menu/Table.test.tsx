import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Menu } from "./Menu";
import { mockData } from "./Menu.mock";

jest.mock("components/atoms/SVG/SVG", () => ({
  SVG: "svg",
}));

describe("Menu", () => {

  
  it("should render the Menu", () => {

    const subject = shallow(<Menu items={[
      {
        icon: "home",
        uri: "/",
      },]}  onChangeRoute= {() => {}} currentPath="" />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {

    const subject = renderer.create(<Menu  items={[
      {
        icon: "home",
        uri: "/",
      },]}  onChangeRoute= {() => {}} currentPath=""/>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
