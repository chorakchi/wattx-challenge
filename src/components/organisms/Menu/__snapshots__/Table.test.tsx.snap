// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly and match to the Snapshot 1`] = `
<div
  className="Boxstyled__BoxStyled-sc-12nhtob-0 bOPHwF Menustyled__StyledWrapper-sc-1vjf9c7-0 gkofae"
>
  <div
    className="Menustyled__StyledItem-sc-1vjf9c7-1 dUQqyy"
    onClick={[Function]}
    selected={true}
  >
    <svg
      name="home"
      size="medium"
    />
  </div>
  <div
    className="Menustyled__StyledGap-sc-1vjf9c7-2 ddQrIZ"
  />
</div>
`;
